scripts/o3de/o3de/manifest.py

```diff
@@ -424,69 +423,53 @@ def get_templates_for_gem_creation():
     return gem_templates
 
 
-def get_templates_for_generic_creation():  # temporary until we have a better way to do this... maybe template_type element
-    def filter_project_and_gem_templates_out(template_path,
-                                             templates_for_project_creation = get_templates_for_project_creation(),
-                                             templates_for_gem_creation = get_templates_for_gem_creation()):
+def get_templates_for_generic_creation(project_path: pathlib.Path = None) -> list:
+    generic_templates = []
+    for template_path in get_all_templates(project_path):
         template_path = pathlib.Path(template_path)
-        return template_path not in templates_for_project_creation and template_path not in templates_for_gem_creation
+        template_json_path = template_path / 'template.json'
+        if not validation.valid_o3de_template_json(template_json_path):
+            continue
+        gem_json_path = template_path / 'Template' / 'gem.json'
+        project_json_path = template_path / 'Template' / 'project.json'
+        if not gem_json_path.is_file() and not project_json_path.is_file():
+            generic_templates.append(template_path)
 
-    return list(filter(filter_project_and_gem_templates_out, get_all_templates()))
+    return generic_templates
```

Comment:
This function was perfectly fine before and this is just copy-and-pasting existing code